src/SmartTransactionsController.ts

```diff
@@ -364,41 +429,36 @@ export default class SmartTransactionsController extends StaticIntervalPollingCo
   #updateSmartTransaction(
     smartTransaction: SmartTransaction,
     {
-      chainId = this.config.chainId,
+      chainId = this.#chainId,
     }: {
       chainId: Hex;
     },
   ) {
-    const { smartTransactionsState } = this.state;
-    const { smartTransactions } = smartTransactionsState;
+    const {
+      smartTransactionsState: { smartTransactions },
+    } = this.state;
     const currentSmartTransactions = smartTransactions[chainId] ?? [];
     const currentIndex = currentSmartTransactions?.findIndex(
       (stx) => stx.uuid === smartTransaction.uuid,
     );
+
     if (currentIndex === -1) {
       return; // Smart transaction not found, don't update anything.
     }
-    this.update({
-      smartTransactionsState: {
-        ...smartTransactionsState,
-        smartTransactions: {
-          ...smartTransactionsState.smartTransactions,
-          [chainId]: smartTransactionsState.smartTransactions[chainId].map(
-            (existingSmartTransaction, index) => {
-              return index === currentIndex
-                ? { ...existingSmartTransaction, ...smartTransaction }
-                : existingSmartTransaction;
-            },
-          ),
-        },
-      },
+
+    this.update((state) => {
+      const currentTransaction = cloneDeep(
+        state.smartTransactionsState.smartTransactions[chainId][currentIndex],
+      );
+      state.smartTransactionsState.smartTransactions[chainId][currentIndex] =
+        Object.assign(currentTransaction, smartTransaction);
     });
```

Comment:
nit: do you think we can make this more idimatic?
```suggestion
      state.smartTransactionsState.smartTransactions[chainId][currentIndex] = {
        ...state.smartTransactionsState.smartTransactions[chainId][
          currentIndex
        ],
        ...smartTransaction,
      };
    });
```

Reply:
This is what I have done in the beginning check the first commit 
Then the ci security check flagged this as "Prototype-polluting assignment"

Reply:
To fix the "prototype-polluting assignment" error, we have the `isSafeDynamicKey` utility in controller-utils (see https://github.com/MetaMask/core/pull/4041).

```suggestion
      const currentTransaction = cloneDeep(
        state.smartTransactionsState.smartTransactions[chainId][currentIndex],
      );
      if (
        [chainId, currentIndex].every((key) => isSafeDynamicKey(String(key)))
      ) {
        state.smartTransactionsState.smartTransactions[chainId][currentIndex] =
          Object.assign(currentTransaction, smartTransaction);
      }
    });
```

Reply:
just using cloneDeep and Objet.assign did fix it I believe ?

Reply:
I'm still seeing the warning in this PR.